serve/mlc_serve/engine/base.py

```diff
@@ -19,10 +20,8 @@
 class RawLogprobsInfo:
     current_token_id: int
     current_logprob: float
-    top_token_ids: Optional[np.ndarray]
-    top_logprobs: Optional[np.ndarray]
-
-RawLogprobsInfos = List[Optional[RawLogprobsInfo]]
```

Comment:
Some words about `RawLogprobsInfos`: it has been removed, but after my comment replaced by List[RawLogprobsInfo]. I've tried to follow pipeline with it and I saw that `Optional` was missed, I found it in `SampleOutput` only. Why we need it, due to we do not have any separator which splits requests with and without logprobs, it means that in general case we have mix of None and `RawLogprobsInfo` in the list.
We did not catch it in the tests due to `--logprobs` add logprobs in all requests.
I think it is better:
1. add `Optional` or return RawLogprobsInfos
2. add test for mix of requests with and without logprobs

Reply:
Added `Optional` for now. My main complaint with `RawLogprobsInfos` is that it is very easy to get confused with `RawLogprobsInfo` unless you look very carefully. I'm okay with introducing new variable, but something more clear. I hope we can rename these variables more intuitive in general, but I do not have a good idea now. So, leave it as it is, please feel free to follow-up if you have any better idea. 

I also added the new test for mix of requests.